.github/workflows/main.yml

```diff
@@ -0,0 +1,153 @@
+name: A workflow for updating discovery artifacts
+# Controls when the action will run.
+
+on:
+  # Triggers the workflow on pull request events but only for the master branch
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Update Discovery Artifacts PR
+    runs-on: ubuntu-latest
+    if: ${{github.event.pull_request.user.login == 'parthea' && github.event.pull_request.title == 'Add github action to update discovery artifacts'}}
+    steps:
+      - name: Check out working directory for changes
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/master
+          path: ./branch
+          fetch-depth: '0'
+
+      - name: Check out branch with scripts
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/add-github-action-to-update-discovery-artifacts
+          path: ./branch_with_scripts
+          fetch-depth: '0'
+
+      - name: Check out master
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/master
+          path: ./master
+
+      - name: Create branch
+        run: |
+          git checkout -b update-discovery-artifacts
+        working-directory: ./branch
+
+      - name: Clone discovery artifacts from discovery-artifact manager
+        uses: actions/checkout@v2
+        with:
+          repository: googleapis/discovery-artifact-manager
+          path: ./discovery-artifact-manager
+
+      - name: Copy scripts
+        run: |
+          cp ../branch_with_scripts/changesummary.py changesummary.py
+          cp ../branch_with_scripts/create_commits.sh create_commits.sh
+        working-directory: ./branch
+
+      - name: Copy discovery artifacts
+        run: |
+          cp ../discovery-artifact-manager/discoveries/*.json googleapiclient/discovery_cache/documents/
+        working-directory: ./branch
+
+      - name: Get changed discovery artifacts
+        run: git diff origin/master --name-only -- '*.json' |
+              sed 's/.*\///' > changed_files
+        working-directory: ./branch
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+
+      - name: Install pandas
+        run: pip3 install pandas
```

Comment:
Could you create a `requirements.txt` and pin this to a specific version? It should make it easier to track down any issues stemming from new pandas releases.

Reply:
Done.